code/modules/power/overmap_solar.dm

```diff
@@ -0,0 +1,206 @@
+#define SOLAR_MAX_DIST 40
+
+// This is just to keep all the overmap solar stuff nice and tidy and not spread across three or four files.
+
+//
+// Overmap solars below. It'd be an absolutely pain to add a bunch of if statements and such to try and shoehorn normal function into them.
+//
+
+
+// SOLAR PANELS
+
+/obj/machinery/power/solar/overmap
+	var/obj/machinery/power/solar_control/overmap/overmap_control = null
+
+//calculates the fraction of the sunlight that the panel recieves
+/obj/machinery/power/solar/overmap/update_solar_exposure()
+	if(obscured)
+		sunfrac = 0
+		return
+	//find the smaller angle between the direction the panel is facing and the direction of the sun (the sign is not important here)
+	var/p_angle = min(abs(adir - overmap_control.overmap_tracker.get_sun_angle()), 360 - abs(adir - overmap_control.overmap_tracker.get_sun_angle()))
+
+	if(p_angle > 90)			// if facing more than 90deg from sun, zero output
+		sunfrac = 0
+
+		return
+	//Do distance checks from the star, keep sunfrac at a min of 0 and a max of 1.
+	var/obj/effect/overmap/star/star = locate(/obj/effect/overmap/star)
+	var/dist = get_dist(star, overmap_control.linked)
+	var/dist_penalty = dist * 0.1
+
+	sunfrac = cos(p_angle) ** 2
+	sunfrac -= (dist_penalty -= star.star_luminosity)
+	//Opacity check.
+	var/list/turflist = getline(overmap_control.linked, star)
+	for(var/turf/T in turflist)
+		var/obj/effect/overmap/overmap_effect = locate(/obj/effect/overmap) in T
+		if(overmap_effect && overmap_effect.opacity)
+			sunfrac = 0
+			break
+	if(!star.star_luminosity)
+		sunfrac = 0
+
+	sunfrac = Clamp(sunfrac, 0, 1)
+
+/obj/machinery/power/solar/overmap/unset_control()
+	if(overmap_control)
+		overmap_control.connected_panels.Remove(src)
+	overmap_control = null
+
+/obj/machinery/power/solar/overmap/Process()
+	if(stat & BROKEN)
+		return
+	if(!overmap_control) //if there's no sun or the panel is not linked to a solar control computer, no need to proceed
+		return
+
+	if(powernet)
+		if(powernet == overmap_control.powernet)//check if the panel is still connected to the computer
+			if(obscured) //get no light from the sun, so don't generate power
+				return
+			var/sgen = solar_gen_rate * sunfrac * efficiency
+			add_avail(sgen)
+			overmap_control.gen += sgen
+		else //if we're no longer on the same powernet, remove from control computer
+			unset_control()
+
+//set the control of the panel to a given computer if closer than SOLAR_MAX_DIST
+/obj/machinery/power/solar/overmap/set_control(var/obj/machinery/power/solar_control/overmap/SC)
+	if(SC && (get_dist(src, SC) > SOLAR_MAX_DIST))
+		return 0
+	overmap_control = SC
+	return 1
+
+//set the control of the panel to null and removes it from the control list of the previous control computer if needed
+/obj/machinery/power/solar/overmap/unset_control()
+	if(overmap_control)
+		overmap_control.connected_panels.Remove(src)
+	overmap_control = null
```

Comment:
Nit: you check if it's false and then null it regardless, could indent this one step.